server/models/video/video.ts

```diff
@@ -1706,15 +1706,23 @@ export class VideoModel extends Model<Partial<AttributesOnly<VideoModel>>> {
   }
 
   getFormattedVideoFilesJSON (includeMagnet = true): VideoFile[] {
+    return videoFilesModelToFormattedJSON(this, this.VideoFiles, { includeMagnet })
+  }
+
+  getFormattedVideoStreamingPlaylistsJSON (includeMagnet = true): VideoFile[] {
+    return flatten(this.VideoStreamingPlaylists.map(p => videoFilesModelToFormattedJSON(this, p.VideoFiles, { includeMagnet })))
+  }
+
+  getFormattedVideoFilesWithStreamingPlaylistsJSON (includeMagnet = true): VideoFile[] {
```

Comment:
I had to separate out `getFormattedVideoFilesJSON` to avoid deduplicating videos after the query.  We need to get all video files and this seemed like the best approach.  Otherwise I'd just be duplicating the logic of [videoFilesModelToFormattedJSON](https://github.com/Chocobozzz/PeerTube/blob/67599b83c544e05d62112e1faed808645a8e4303/server/models/video/formatter/video-format-utils.ts#L248C10-L289).

I believe you had suggested to use [getMaxQualityFile](https://github.com/Chocobozzz/PeerTube/blob/0c9668f77901e7540e2c7045eb0f2974a4842a69/server/models/video/video.ts#L1625-L1627) but that only gets the highest quality and not all available files.